File: examples/src/Shapes/AccessOLEObjectFrame.cs
using System.IO
import aspose.slides as slides

namespace Aspose.slides.Examples.CSharp.shapes
{
    public class AccessOLEObjectFrame
    {
        public static void Run()
        {
            #ExStart:AccessOLEObjectFrame
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Shapes()

            # Load the PPTX to Presentation object
            using (Presentation pres = new Presentation(dataDir + "AccessingOLEObjectFrame.pptx"))
            {
                # Access the first slide
                sld = pres.slides[0]

                # Cast the shape to OleObjectFrame
                OleObjectFrame oleObjectFrame = sld.shapes[0] as OleObjectFrame

                # Read the OLE Object and write it to disk
                if (oleObjectFrame != None)
                {
                    # Get embedded file data
                    byte[] data = oleObjectFrame.EmbeddedData.EmbeddedFileData

                    # Get embedded file extention
                    fileExtention = oleObjectFrame.EmbeddedData.EmbeddedFileExtension

                    # Create path for saving the extracted file
                    extractedPath = dataDir + "excelFromOLE_out" + fileExtention

                    # Save extracted data
                    using (FileStream fstr = new FileStream(extractedPath, FileMode.Create, FileAccess.Write))
                    {
                        fstr.Write(data, 0, data.Length)
                    }
                }
            }

            #ExEnd:AccessOLEObjectFrame
        }
    }
}

File: examples/src/Shapes/AccessOLEObjectFrame.py
import aspose.slides as slides


def shapes_accessing_ole_object_frame():
    dataDir = "./examples/data/"
    outDir = "./examples/out/"

    # Load the PPTX to Presentation object
    with slides.Presentation(dataDir + "shapes_accessing_ole_object_frame.pptx") as pres:
        # Access the first slide
        sld = pres.slides[0]

        # Cast the shape to OleObjectFrame
        oleObjectFrame = sld.shapes[0]

        # Read the OLE Object and write it to disk
        if type(oleObjectFrame) is slides.OleObjectFrame:
            # Get embedded file data
            data = oleObjectFrame.embedded_data.embedded_file_data

            # Get embedded file extention
            fileExtention = oleObjectFrame.embedded_data.embedded_file_extension

            # Create a path to save the extracted file
            extractedPath = "excelFromOLE_out" + fileExtention

            # Save extracted data
            with open(outDir + "shapes_accessing_ole_object_frame_out.xlsx", "wb") as fs:
                fs.write(data)